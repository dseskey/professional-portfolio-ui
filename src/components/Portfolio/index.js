import React, { useState } from 'react';
import { Container, Row, Col, Card } from 'react-bootstrap';
import projectJson from '../../assets/json/projects.json';
import Project from '../Project';
function Portfolio() {

    return (
        <Container fluid data-testid='portfolio-container' id='portfolio-container' className='pagesParentContainer'>
            <Row className="pageHeader">
                <h2 data-testid='portfolio-section-header'>My Projects</h2>
            </Row>
            <Row data-testid='projects-card-row' id='projects-card-row'>
                {projectJson.projects.map((project) => {
                    return (
                        <Project key={`${project.title}-project`} project={project} />
                    )
                })}
            </Row>

        </Container >
    )
}

export default Portfolio;