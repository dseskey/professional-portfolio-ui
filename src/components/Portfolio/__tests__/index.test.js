import React from 'react';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Portfolio from '..'

afterEach(cleanup);


describe("Project component", () =>{
    it('renders', () => {
        render(<Portfolio/>);
    });
    it('matches snapshot DOM node structure', () => {
        const {asFragment} = render(<Portfolio/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('has a section container rendered with a portfolio-container ID and the required class of pagesParentContainer', () => {
       const {getByTestId} = render(<Portfolio/>);
       expect(getByTestId('portfolio-container')).toHaveClass('pagesParentContainer');
       expect(getByTestId('portfolio-container')).toHaveAttribute('id', 'portfolio-container');

    });
    
    it('has a card for the Travelers Mobile application', () => {
        const {getByTestId} = render(<Portfolio/>);
        expect(getByTestId('Travelers Mobile-card')).toHaveClass('card');
     });

     it('has a project-cards-row with 4 projects (children nodes that represent cards)', () => {
        const {getByTestId} = render(<Portfolio/>);
        expect(getByTestId('projects-card-row')).toHaveClass('row');
        expect(getByTestId('projects-card-row').childElementCount).toEqual(4);

     });

});

