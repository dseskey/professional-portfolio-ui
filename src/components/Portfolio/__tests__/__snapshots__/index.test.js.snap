// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="pagesParentContainer container-fluid"
    data-testid="portfolio-container"
    id="portfolio-container"
  >
    <div
      class="pageHeader row"
    >
      <h2
        data-testid="portfolio-section-header"
      >
        My Projects
      </h2>
    </div>
    <div
      class="row"
      data-testid="projects-card-row"
      id="projects-card-row"
    >
      <div
        class="col-md-4"
        data-testid="Travelers Mobile-col"
      >
        <div
          class="card"
          data-projectname="Travelers Mobile"
          data-testid="Travelers Mobile-card"
          style="background-image: url(https://s3.us-east-2.amazonaws.com/dseskey.com/images/professional-portfolio/trvMobileApp.jpeg);"
        />
      </div>
      <div
        class="col-md-4"
        data-testid="Noggle Boggle Trivia and Quiz Generator-col"
      >
        <div
          class="card"
          data-projectname="Noggle Boggle Trivia and Quiz Generator"
          data-testid="Noggle Boggle Trivia and Quiz Generator-card"
          style="background-image: url(https://s3.us-east-2.amazonaws.com/dseskey.com/images/professional-portfolio/IMG_1168.jpeg);"
        />
      </div>
      <div
        class="col-md-4"
        data-testid="Single Plant Irrigation System (SPIS)-col"
      >
        <div
          class="card"
          data-projectname="Single Plant Irrigation System (SPIS)"
          data-testid="Single Plant Irrigation System (SPIS)-card"
          style="background-image: url(https://s3.us-east-2.amazonaws.com/dseskey.com/images/professional-portfolio/spis.jpg);"
        />
      </div>
      <div
        class="col-md-4"
        data-testid="Photography Portfolio-col"
      >
        <div
          class="card"
          data-projectname="Photography Portfolio"
          data-testid="Photography Portfolio-card"
          style="background-image: url(https://s3.us-east-2.amazonaws.com/dseskey.com/images/professional-portfolio/photo-port.jpg);"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
